### PR TITLE
Completed Friendship v1 Specifications

### PR DESCRIPTION
**Pull request by chasscepts**
We implemented the following features in this milestone
- We placed a link against all users' names in the users' list.
- The link is not shown on the current user or any of their friend's (both confirmed and pending) names.
- On clicking the link, a friend request is sent to that user and they can see it and any other pending requests on their page. 
- They can choose to accept or reject the request. 
- If they reject the request, it is deleted from the database.
- If they accept the request, it is marked as confirmed.
- The current user can see both their and their friends' posts on their page.

### CODE REVIEW

**Review by ElijahTobs — CHANGES_REQUESTED** [review]
# Status: Changes Required :recycle: 

## Hi @chasscepts and @iliebabcenco  :wave: 

You've done well so far :+1:  and I commend you for that :clap: 
However, there are still some changes to be made before this milestone can be approved.

## REQUIRED

* [x] There should be a button to invite user to friendship on both users list page and single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) but the button is only present on the users list page.
* [x] Posts timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) but your posts timeline include posts by non-friend users (i.e users who are not friends)
* [ ] When `User A` sends an invite to `User B` using the `Send Invite` button, the `Accept` and `Reject` button should be available for `User B` but there are no buttons. In other words, a user is not able to see friendship requests. Kindly see to that.


## NICE TO HAVE
* [ ] When `User A` sends a friend request to `User B`, you can replace the _"Send Invite"_ button with a text _"Friend request sent"_

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Comment by chasscepts** [comment]
We have fixed points 1 and 2 raised in the first review. Point 3 was already implemented before we requested for code review. A user can see all pending friend requests when they are on their page. Please see the image below.

![[]](/iliebabcenco/ror-social-scaffold/blob/friendship1/docs/pending_friends.png?raw=true)

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -17,6 +17,11 @@
         <div class="menu-item">
           <%= menu_link_to 'All users', users_path %>
         </div>
+        <% unless current_user.nil? %>
+          <div class="menu-item">
+            <%= menu_link_to 'My Page', user_path(current_user) %>
+          </div>
+        <% end %>
```
Comment (SajjadAhmad14) [inline app/views/layouts/application.html.erb:24]: Please move this logic to an halper.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -17,6 +17,11 @@
...
       </div>
       <% if current_user %>
```
Comment (SajjadAhmad14) [inline app/views/layouts/application.html.erb:26]: Please move this logic to an halper.

File: app/views/users/_user.html.erb [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+  <div class="profile-link-wrap">
+    <%= link_to 'View Profile',  user_path(user), class: 'profile-link' %>
+  </div>
+  <div class='invite-friend-form'>
+    <% if include_invite_link?(current_user, user) %>
+      <%= form_for(current_user.initiated_friendships.new, url: user_friendships_path(current_user)) do |form| %>
+        <%= form.hidden_field :invitee_id, value: user.id %>
+        <%= form.submit 'Send Invite', class: 'link-btn' %>
+      <% end %>
+    <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/_user.html.erb:12]: Please move this logic to an halper.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
+                        <%= form.submit 'Accept', class: 'link-btn' %>
+                        <% end %>
+                    </div>
+                    <%= link_to 'Reject', reject_friendship_path(friend), class: 'link-control'  %>
+                </div>
+                <hr>
+                <% end %>
+            </div>
+        </aside>
+        <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb:36]: Please move this logic to a helper.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Invalid Request
### **Friendly reminder:** "You should use Git Flow in this project. Please take a step back and go through this [lesson](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) and watch the video in it to understand the flow. Then create a brand new repo with the correct setup and feature branch (with DESCRIPTIVE name) and copy your code there or close this pull request and create another one with the correct feature branch. This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!"

## Pull Request should be made from the `feature branch` to the `development` branch and the `feature branch` has some descriptive name.

### Happy Coding ☺️

## You can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request closed without merging**